tests/auth_plugin_test.php

```diff
@@ -614,6 +614,14 @@ public function test_get_request_login_url_user_parameters_based_on_plugin_confi
             'lastname'  => new external_value(PARAM_NOTAGS, 'The family name of the user', VALUE_OPTIONAL),
             'email'     => new external_value(PARAM_RAW_TRIMMED, 'A valid and unique email address', VALUE_OPTIONAL),
             'username'  => new external_value(PARAM_USERNAME, 'A valid and unique username', VALUE_OPTIONAL),
+
+            'department'  => new external_value(PARAM_TEXT, 'Department of the user', VALUE_OPTIONAL),
```

Comment:
Those fields are still hardcoded. Can you please modify the code to be able to provide a list of any fields. An example of that in core could be find here https://github.com/moodle/moodle/blob/master/user/externallib.php#L105

So we would have similar here e.g. "extrafields" multiple structure with "name" and "value". There should be a validation before processing and If we receive an unknown field we should throw an exception.

Standard user fields should be a good start. In a future we can improve it to process custom userfields as well.

Also we would like to see unit tests with any of the code changes.